src/handler.ts

```diff
@@ -0,0 +1,195 @@
+import * as assert from "assert";
+import { APIGatewayProxyHandler } from "aws-lambda";
+import { DynamoDB } from "aws-sdk";
+import * as uuid from "uuid/v4";
+import getConfig from "./getConfig";
+import { DocumentClient } from "aws-sdk/clients/dynamodb";
+
+type Container = {
+  config: {
+    DynamoDB: object;
+    stage: string;
+  };
+  dynamodb: DynamoDB;
+  client: DocumentClient;
+  TableName: string;
+};
+const retrieveContainer: () => Container = () => {
+  const config = getConfig();
+  const dynamodb = new DynamoDB({ ...config.DynamoDB });
+
+  return {
+    config,
+    dynamodb,
+    client: new DynamoDB.DocumentClient({ service: dynamodb }),
+    TableName: `${config.stage}-User`
+  };
+};
+
+const getUser = async (userId: string, { client, TableName }: Container) => {
+  const dynamoParams = {
+    ExpressionAttributeValues: {
+      ":id": userId
+    },
+    KeyConditionExpression: "id = :id",
+    TableName
+  };
+
+  const result = await client.query(dynamoParams).promise();
+
+  return result.Items && result.Items.length && result.Items[0];
+};
+
+const respondWithUser = async (userId: string, container: Container) => {
+  const user = await getUser(userId, container);
+
+  assert(user, "User Not Found");
+
+  return {
+    statusCode: 200,
+    body: JSON.stringify(user)
+  };
+};
+
+export const readUser: APIGatewayProxyHandler = async event => {
+  try {
+    const userId = event.pathParameters!.userId;
```

Comment:
Im curious as to why you chose to do a non-null assertion vs typing the event?

Reply:
Good question.  I think it was either trying to be quick to make the type checker happy, or being overly sensitive to the Superformula team potentially not liking the verbosity of typescript?

I'm not actually sure of the best way to type `event.pathParameters`.  Event is an `APIGatewayProxyEvent` found in the `aws-lambda` package.

So I can do something like: 

```
(event.pathParameters as APIGatewayProxyEvent["pathParameters"])
```

But event.pathParameters can still be `null`.

Maybe something like:

```
type PathParameters = { userId: string };
const userId = (event.pathParameters as PathParameters).userId;
```

would be how I would iterate on this.  Curious how you would handle.

Reply:
(I just edited to add syntax highlighting to the code blocks)

Reply:
Thanks for the response on this! I definitely understand that typescript is not the quickest to prototype with so there's always plenty of `!` and `as any` whenever I test things. But there are a few ways you could handle this:

1) In the variable declaration (preferred method)
```ts
export const readUser: APIGatewayProxyHandler = async (event: APIGatewayProxyEvent) => {...}
```
2) Through nested traversal (which is what you had brought up)
```ts
const pathParam = event.pathParameters as APIGatewayProxyEvent["pathParameters"];
const userId = pathParam.userId;
```

Although, It looks like neither of these aren going to solve your problem because looking at their code there's always going to be a null for that param:
```ts
export interface APIGatewayProxyEvent {
    ...
    pathParameters: { [name: string]: string } | null;
    ...
}
```
The real issue is that I think the typing is not super great in the `aws-lambda` package. They should really extend their interface as constructor that can take in variables to create the proper definitions. That way you can inform the interface that you have defined `pathParameters`.

So I think the answer to this would have to be the custom definition solution you went with:
```ts
type PathParameters = { userId: string };
const userId = (event.pathParameters as PathParameters).userId;
```

Or a pull request against the library but that's a little out of scope here.